File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/memento/SaveManager.java
package memento;

import java.util.ArrayList;
import java.util.List;

class SaveManager {
    private final List<GameState> savedStates = new ArrayList<>();

    public void add(GameState gameState) {
        savedStates.add(gameState);
    }

    public GameState get(int index) {
        return savedStates.get(index);
    }
}

File: code/typescript/memento/SaveManager.ts
import { GameState } from "./GameState";

export class SaveManager {
    private savedStates: GameState[] = [];

    add(gameState: GameState): void {
        this.savedStates.push(gameState);
    }

    get(index: number): GameState {
        return this.savedStates[index];
    }
}
